cluster/aks.go

```diff
@@ -189,9 +197,10 @@ func (c *AKSCluster) CreateCluster() error {
 			count := int32(np.Count)
 			name := np.Name
 			profiles = append(profiles, containerservice.ManagedClusterAgentPoolProfile{
-				Name:   &name,
-				Count:  &count,
-				VMSize: containerservice.VMSizeTypes(np.NodeInstanceType),
+				Name:         &name,
+				Count:        &count,
+				VMSize:       containerservice.VMSizeTypes(np.NodeInstanceType),
+				VnetSubnetID: getVNetSubnetID(np),
```

Comment:
Is it a valid scenario to pass in a subnet that belongs to a resource group other than the resource group the cluster is created in? If not than add validation to check that the resource group of the subnet matches `c.GetResourceGroupName()` which is used for creating the cluster.

Reply:
The Azure web interface only lists VNs from the same resource group, so I think no, the subnet must be in the same resource group as the cluster. Also, it has to be under the same subscription. But should we check that? Azure will check these for us if and when it needs to, I don't think we should duplicate the logic here. The UI already provides a way to select a subnet that should work, and users will always be able to shoot themselves in the foot if they really want to.